why-changelog-with-commitizen/index.qmd

```diff
@@ -0,0 +1,222 @@
+changes for each version of a project. It typically includes the version
+number, release date, and a list of the changes made in that version.
+
+However, creating and maintaining a changelog manually can be
+time-consuming and error-prone.Fortunately, many tools can automatically
+generate changelogs based on a project's history. So the question
+becomes:
+
+*Which tool should we use to auto-generate changelogs for our projects?*
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+Motivations for this decision include:
+
+-   Having a consistent format for changelogs makes it easier for users
+    and contributors to understand and keep track of the changes between
+    versions.
+-   We do not want to create changelogs manually because it's
+    time-consuming and error-prone.
+-   Having a tool that can auto-generate changelogs based on commit
+    messages can help enforce good commit and merging practices.
+
+Requirements for the tool include:
+
+-   It should support GitHub Action workflows for auto-generating the
+    changelog
+-   It should generate changelogs following [Semantic
+    Versioning](/why-semver/index.qmd) (SemVer).
+-   It should generate changelogs using commit messages that follow the
+    [Conventional Commits](/why-conventional-commits/index.qmd)
+    specification.
+-   It should be well-documented and actively maintained.
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+There are numerous tools available for automatically generating
+changelogs for Git repositories following the Conventional Commit
+specification, as seen by [this
+list](https://www.conventionalcommits.org/en/about/#tooling-for-conventional-commits)
+by Conventional Commits. We have considered the following options:
+
+-   [Commitizen](https://commitizen-tools.github.io/commitizen/)
+-   [Python Semantic
+    Release](https://python-semantic-release.readthedocs.io/en/latest/index.html)
+-   [git-cliff](https://git-cliff.org/docs/)
+
+### Commitizen
+
+[Commitizen](https://commitizen-tools.github.io/commitizen/) is a
+release management tool designed for teams. By leveraging a standardized
+commit convention (Conventional Commits by default, but this can be
+customised), it streamlines the process of version bumping and
+generating changelogs.
+
+::::: columns
+::: column
+#### Benefits
+
+-   Has GitHub Action workflows to auto-generate the changelog.
+-   Can generate changelogs based on commit messages following the
+    SemVer and Conventional Commits specification.
+-   Is well-documented.
+-   Is actively maintained.
+-   Will be very easy to implement, since we already use it for [linting
+    our commit messages](../why-lint-with-commitizen/index.qmd) and
+    [versioning](../why-semantic-release-with-commitizen/index.qmd).
```

Comment:
```suggestion
-   Will be very easy to implement, since we already use it for [linting
    our commit messages](/why-lint-with-commitizen/index.qmd) and
    [versioning](/why-semantic-release-with-commitizen/index.qmd).
```

Good to be consistent with always using the root website to link internally.